src/internal/connector/graph/service.go

```diff
@@ -99,6 +106,24 @@ func (c *clientConfig) populate(opts ...option) *clientConfig {
 	return c
 }
 
+// apply updates the http.Client with the expected options.
+func (c *clientConfig) applyMiddlewareConfig(retryoptions *RetryHandlerOptions) {
+	if c.retry.NoRetry {
+		retryoptions = &RetryHandlerOptions{NoRetry: true}
+	} else {
+		if c.retry.MaxRetries > 0 {
+			retryoptions.MaxRetries = c.retry.MaxRetries
+		} else {
+			retryoptions.MaxRetries = defaultMaxRetries
+		}
+		if c.retry.Delay > 0 {
+			retryoptions.Delay = c.retry.Delay
+		} else {
+			retryoptions.Delay = defaultDelay
+		}
+	}
```

Comment:
nit: for code cleanliness, prefer returns over nested else blocks.
```suggestion
	if c.retry.NoRetry {
		retryoptions = &RetryHandlerOptions{NoRetry: true}
		return
	}

	retryoptions.MaxRetries = defaultMaxRetries
	if c.retry.MaxRetries > 0 {
		retryoptions.MaxRetries = c.retry.MaxRetries
	}
	
	retryoptions.Delay = defaultDelay
	if c.retry.Delay > 0 {
		retryoptions.Delay = c.retry.Delay
	}
```